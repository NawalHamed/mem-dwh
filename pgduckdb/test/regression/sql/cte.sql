CREATE TABLE t(a INT);

-- Not supported by DuckDB, so we shouldn't send it there
WITH modifying_cte AS (
    INSERT INTO t VALUES (1) RETURNING *
) select * from modifying_cte;

WITH RECURSIVE outermost(x) AS (
 SELECT 1
 UNION (WITH innermost1 AS (
  SELECT 2
  UNION (WITH innermost2 AS (
   SELECT 3
   UNION (WITH innermost3 AS (
    SELECT 4
    UNION (WITH innermost4 AS (
     SELECT 5
     UNION (WITH innermost5 AS (
      SELECT 6
      UNION (WITH innermost6 AS
       (SELECT 7)
       SELECT * FROM innermost6))
      SELECT * FROM innermost5))
     SELECT * FROM innermost4))
    SELECT * FROM innermost3))
   SELECT * FROM innermost2))
  SELECT * FROM outermost
  UNION SELECT * FROM innermost1)
 )
 SELECT * FROM outermost ORDER BY 1;

DROP TABLE t;
