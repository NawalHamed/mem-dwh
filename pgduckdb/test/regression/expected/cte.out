CREATE TABLE t(a INT);
-- Not supported by DuckDB, so we shouldn't send it there
WITH modifying_cte AS (
    INSERT INTO t VALUES (1) RETURNING *
) select * from modifying_cte;
 a 
---
 1
(1 row)

WITH RECURSIVE outermost(x) AS (
 SELECT 1
 UNION (WITH innermost1 AS (
  SELECT 2
  UNION (WITH innermost2 AS (
   SELECT 3
   UNION (WITH innermost3 AS (
    SELECT 4
    UNION (WITH innermost4 AS (
     SELECT 5
     UNION (WITH innermost5 AS (
      SELECT 6
      UNION (WITH innermost6 AS
       (SELECT 7)
       SELECT * FROM innermost6))
      SELECT * FROM innermost5))
     SELECT * FROM innermost4))
    SELECT * FROM innermost3))
   SELECT * FROM innermost2))
  SELECT * FROM outermost
  UNION SELECT * FROM innermost1)
 )
 SELECT * FROM outermost ORDER BY 1;
 x 
---
 1
 2
 3
 4
 5
 6
 7
(7 rows)

DROP TABLE t;
